File: benchmarks/pandas/bench_cat_reorder_as_ordered.py
"""Benchmark: cat_reorder_as_ordered — pandas Categorical reorder_categories/as_ordered/as_unordered on 100k-element Series"""
import json
import time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

cats = ["a", "b", "c", "d"]
s = pd.Categorical([cats[i % len(cats)] for i in range(ROWS)], categories=cats)

for _ in range(WARMUP):
    _ = s.reorder_categories(["d", "c", "b", "a"])
    _ = s.as_ordered()
    _ = s.as_unordered()

start = time.perf_counter()
for _ in range(ITERATIONS):
    _ = s.reorder_categories(["d", "c", "b", "a"])
    _ = s.as_ordered()
    _ = s.as_unordered()
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "cat_reorder_as_ordered", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_cat_reorder_as_ordered.ts
/**
 * Benchmark: cat_reorder_as_ordered — CategoricalAccessor reorderCategories/asOrdered/asUnordered on 100k-element Series
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const cats = ["a", "b", "c", "d"];
const s = new Series({ data: Array.from({ length: ROWS }, (_, i) => cats[i % cats.length]) });

for (let i = 0; i < WARMUP; i++) {
  s.cat.reorderCategories(["d", "c", "b", "a"]);
  s.cat.asOrdered();
  s.cat.asUnordered();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.cat.reorderCategories(["d", "c", "b", "a"]);
  s.cat.asOrdered();
  s.cat.asUnordered();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "cat_reorder_as_ordered",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
